pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -123,46 +161,64 @@ func (r *Reconciler) gatherSelectedResource(placement string, selectors []fleetv
 		}
 	}
 	// sort the resources in strict order so that we will get the stable list of manifest so that
-	// the generated work object doesn't change between reconcile loops
-	sortResources(resources)
+	// the generated work object doesn't change between reconcile loops.
+	sortResources(resources, buildApplyOrderMap())
```

Comment:
Hi Arvind! This is still being called every time resources are selected I am afraid.

Reply:
How about a top-level variable like `var applyOrderMap = buildApplyOrderMap()` and then refer to the variable here?